src/java/org/apache/cassandra/db/Keyspace.java

```diff
@@ -534,7 +534,7 @@ else if (isDeferrable)
                 }
             }
 
-            long acquireTime = currentTimeMillis() - mutation.viewLockAcquireStart.get();
+            long acquireTime = currentTimeMillis() - mutation.viewLockAcquireStart;
```

Comment:
Frankly, it sounds a bit odd to have both `plain` and `AtomicLongFieldUpdater`-driven access at the same time. Does it really cost much to have plain `mutation` read rather than `Mutation.viewLockAcquireStartUpdater.get(mutation)` instead?

In theory, the underlying platform might fail to support atomic CAS for long, so OpenJDK would resort to `synchronized...` for `compareAndSet`, thus it would be somewhat invalid to use plain reads in those cases.

Reply:
the idea of this change is to reduce allocation by remove AtomicLong in Mutation object (which we create for each request) and replace it with a volatile long + Updater usage

Reply:
agree, there is a cost: https://shipilev.net/blog/2015/faster-atomic-fu/
but for the current scenario it is not that critical - usually we do the operation only once per mutation.